screens/ProfileScreen.tsx

```diff
@@ -31,79 +37,246 @@ export default function ProfileScreen({ navigation }: RootTabScreenProps<'TabOne
     navigation.navigate('LoginRegister')
   })
 
-  return (
-    <View style={styles.container}>
-      {user && (
-        <View style={styles.profileContainer}>
-          <View style={styles.userImageContainer}>
-            <Image source={{ uri: user.photoURL! }} style={styles.userImage} />
+  useEffect(() => {
+    // init temp images
+    //  TODO: init this array using saved images
+    let items = Array.apply(null, Array(60)).map((v, i) => {
+      return {
+        id: i,
+        src: `http://placehold.it/${GRID_IMG_WIDTH}x${GRID_IMG_HEIGHT}?text=` + (i + 1),
+        // src: `https://picsum.photos/${GRID_IMG_WIDTH}/${GRID_IMG_HEIGHT}`,   // an actual image placeholder
+      }
+    })
+    setDataSource(items)
+
+    // TODO: set from persistent storage
+    setCoverPicSrc(require('../assets/images/temp/cover_photo_temp.jpg'))
+
+    // TODO: set from persistent storage
+    setNumCreations(60)
+  }, [])
+
+  const updateCoverPhoto = () => {
+    console.log('Update Cover Photo')
+  }
+
+  // expand image selected from grid
+  const expandImageView = (id: number) => {
+    console.log(`Expand grid image ${id}`)
+  }
+
+  if (user) {
+    return (
+      <View style={styles.container}>
+        <View style={styles.coverRegion}>
+          <ImageBackground resizeMode="cover" source={coverPicSrc as any} style={styles.coverImage}>
+            <View style={styles.IconButtonContainer}>
+              <MaterialCommunityIcons
+                name="logout-variant"
+                size={22}
+                color="black"
+                onPress={onPressLogout}
+                style={styles.logoutButton}
+                title="Logout"
+                accessibilityLabel="Logs the user out"
+              />
+            </View>
+            <View style={styles.IconButtonContainer}>
+              <FontAwesome5
+                name="images"
+                size={20}
+                color="black"
+                onPress={updateCoverPhoto}
+                style={styles.updateCoverPhotoButton}
+                title="Update Cover Photo"
+                accessibilityLabel="Updates the user's cover photo"
+              />
+            </View>
+          </ImageBackground>
+        </View>
+        <View style={styles.content}>
+          <View style={styles.profileHeader}>
+            {user && (
+              <Text style={styles.googleName} numberOfLines={1}>
+                {user.displayName}
+              </Text>
+            )}
+            {user && (
+              <View style={styles.profileImgContainer}>
+                <Image source={{ uri: user.photoURL! }} resizeMode="cover" style={styles.profileImg} />
+              </View>
+            )}
```

Comment:
Nit: No need to check if user is not null again since you already check it in the if statement above.